splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyActionIT.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2012 - 2020 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.foreignkeys;
+
+import com.splicemachine.derby.test.framework.SpliceSchemaWatcher;
+import com.splicemachine.derby.test.framework.SpliceWatcher;
+import com.splicemachine.derby.test.framework.TestConnection;
+import com.splicemachine.test_dao.TableDAO;
+import com.splicemachine.util.StatementUtils;
+import org.junit.*;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.sql.Statement;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Foreign key tests for referential actions:
+ *
+ * ON DELETE NO ACTION
+ * ON DELETE CASCADE
+ * ON DELETE SET NULL
+ * ON UPDATE NO ACTION
+ */
+public class ForeignKeyActionIT {
```

Comment:
Can you test it with "permissive" imports? If we allow up to N badRecords and some records violate the FK constraint, we should log them but import the rest, does that work?

Reply:
I tried to run the following scenario via a unit test but it failed, I suspected it is not related to this PR so I ran it against master and ran into the same issue:

```
create table PI(col1 int, col2 varchar(2), primary key (col1));
create table C1I(col1 int, col2 int, constraint ci_fk_key foreign key(col1) references PI(col1) on delete no action);
insert into PI values (1, 'a');
insert into PI values (2, 'b');
-- create a file: /tmp/bad.csv containing the following rows:
-- 1,1
-- 1000,1000
-- 1,1
-- 2000,2000
-- 2,2
-- create a directory /tmp/BAD
call SYSCS_UTIL.import_data('SPLICE', 'C1I', null, '/tmp/bad.csv', ',', null, null, null, null, 2, '/tmp/BAD', null, null);
rowsImported        |failedRows          |files      |dataSize            |failedLog
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
3                   |2                   |1          |32                  |/tmp/BAD
select * from C1I;
COL1       |COL2
-----------------------
1          |1
1000       |1000
1          |1
2000       |2000
2          |2
```

The `SYSCS_UTIL.IMPORT_DATA` is not rejecting the rows that violate the FK constraint. I will open a JIRA for it so we can track it separately.